server/rest/src/main/java/org/infinispan/rest/server/InfinispanResponse.java

```diff
@@ -0,0 +1,153 @@
+package org.infinispan.rest.server;
+
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.rest.server.operations.mediatypes.MediaType;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpUtil;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.handler.codec.http2.HttpConversionUtil;
+
+public class InfinispanResponse {
```

Comment:
@danberindei It's only a matter of taste in my opinion. Either using setters or `Optional`s has the same result in Netty response. However if there are solid numbers behind your words (e.g. tough data from JMH or Flight Recorder) I will remove Optionals :)